governator-jersey/src/main/java/com/netflix/governator/guice/jetty/GovernatorServletContainer.java

```diff
@@ -0,0 +1,92 @@
+package com.netflix.governator.guice.jetty;
+
+import java.util.Map;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import javax.servlet.ServletException;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.inject.Injector;
+import com.google.inject.Scopes;
+import com.sun.jersey.api.core.ResourceConfig;
+import com.sun.jersey.guice.JerseyServletModule;
+import com.sun.jersey.guice.spi.container.servlet.GuiceContainer;
+import com.sun.jersey.spi.container.servlet.WebConfig;
+
+/**
+ * Enhancement to {@link GuiceContainer} by allowing the Jersey to be configured via bindings instead of 
+ * statically in a {@link JerseyServletModule}.  To enable just provide the following binding,
+ * 
+ * ```java
+ * bind(GuiceContainer.class).to(GovernatorServletContainer.class).asEagerSingleton();
+ * ```
+ * 
+ * To set up a Jersey endpoint,
+ * 
+ * ```java
+ * serve("/*").with(GuiceContainer.class);  
+ * ```
+ * 
+ * To customize the configuration of the Jersey endpoint
+ * 
+ * ```java
+ * @Provides
+ * ResourceConfig getResourceConfig() {
+ *     return new PackagesResourceConfig(ImmutableMap.<String, Object>builder()
+ *             .put(PackagesResourceConfig.PROPERTY_PACKAGES, "org.example.resources")
+ *             .put(ResourceConfig.FEATURE_DISABLE_WADL, "false")
+ *             .build());
+ * }
+ * ```
+ * 
+ * In addition installing this module will automatically turn on the WADL (which can be turned off
+ * by setting the property, com.sun.jersey.config.feature.DisableWADL=true.  Also, when running
+ * in Stage.DEVELOPMENT all scanning singleton resources will be eagerly created.
```

Comment:
This point is a major behavioural change, so would be good to document this new behaviour at the top of the comments block instead of as an afterthought here.